crates/bevy_ecs/src/reflect.rs

```diff
@@ -172,6 +172,15 @@ impl ReflectComponent {
     pub fn new(fns: ReflectComponentFns) -> Self {
         Self(fns)
     }
+
+    /// The underlying function pointers implementing methods on `ReflectComponent`.
+    ///
+    /// Prefer using the methods over fetching the function pointers.
+    /// However, this is useful if you want to cache the functions rather than
+    /// repetitively using `TypeRegistry::get()` followed by `data()`.
```

Comment:
Nit: I feel like this should be less about "caching vs `TypeRegistry::get()`" and more about just needing a subset of these function pointers directly. We want users to know that they are free to clone `ReflectComponent` and `ReflectResource` if they want to cache them (no need to `TypeRegistry::get()` every time). But, like in your case, this method becomes useful when you want to cache only a subset of these pointers and not necessarily all of them.